apps/gov-website/pages/proposal/new.tsx

```diff
@@ -1,168 +1,86 @@
 import type { GetStaticProps, NextPage } from "next";
-import { FormEventHandler, useCallback } from "react";
-import { If } from "react-extras";
+import { ChangeEventHandler, useCallback, useEffect, useState } from "react";
 
 import {
 	extractCallableExtrinsics,
 	getApiInstance,
 } from "@app-gov/service/cennznet";
 import { CENNZ_NETWORK } from "@app-gov/service/constants";
 import {
-	AccountSelect,
-	Button,
 	FunctionCallFieldSet,
 	Header,
 	Layout,
-	MarkdownField,
-	Select,
+	ProposalNewForm,
+	ProposalNewFormDialog,
+	useTransactionDialog,
 } from "@app-gov/web/components";
-import { useControlledCheckbox, useControlledInput } from "@app-gov/web/hooks";
+import { useProposalNewForm } from "@app-gov/web/hooks";
 
-interface StaticProps {
+interface NewProposalProps {
 	extrinsics: Awaited<ReturnType<typeof extractCallableExtrinsics>>;
 }
 
-export const getStaticProps: GetStaticProps<StaticProps> = async () => {
+export const getStaticProps: GetStaticProps<NewProposalProps> = async () => {
 	const api = await getApiInstance(CENNZ_NETWORK.ChainSlug);
 	const extrinsics = await extractCallableExtrinsics(api);
-
 	return {
 		props: {
 			extrinsics,
 		},
 	};
 };
 
-const NewProposal: NextPage<StaticProps> = ({ extrinsics }) => {
-	const { value: justification, onChange: onJustificationChange } =
-		useControlledInput<string, HTMLTextAreaElement>("");
+const NewProposal: NextPage<NewProposalProps> = ({ extrinsics }) => {
+	const { submitForm, formState } = useProposalNewForm();
+	const [formKey, setFormKey] = useState<string>(`PrposalNewForm${Date.now()}`);
```

Comment:
React has a new hook `useId` for this purpose, though I'm not sure how you'd go about resetting it - worth a look

https://reactjs.org/docs/hooks-reference.html#useid

Reply:
yeah good call, but looks like it's for assigning an id to element, not for key